web/application.html

```diff
@@ -0,0 +1,209 @@
+<!doctype html>
+<html lang="en-us">
+<head>
+    <meta charset="utf-8">
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+    <title>Urho3D</title>
+    <style>
+        html, body {
+            padding: 0px;
+            margin: 0px;
+            margin: 0px;
+            overflow: hidden;
+        }
+        .emscripten {
+            padding-right: 0;
+            margin-left: auto;
+            margin-right: auto;
+            display: block;
+        }
+        div.emscripten {
+            text-align: center;
+        }
+        .fulscreen-container {
+            width: 100%;
+            position: absolute;
+            text-align: center;
+        }
+        div.emscripten_border {
+            border: none;
+        }
+        #canvas {
+            position: absolute;
+            top: 0;
+            left: 0;
+            width: 100%;
+            height: 100%;
+            background-color: #000;
+            border: 0px;
+            cursor: pointer !important;
+        }
+        canvas.emscripten {
+            border: none;
+        }
+        .centered {
+            position: absolute;
+            top: 50%;
+            left: 50%;
+            -moz-transform: translateX(-50%) translateY(-50%);
+            -webkit-transform: translateX(-50%) translateY(-50%);
+            transform: translateX(-50%) translateY(-50%);
+        }
+    </style>
+</head>
+<body>
+    <div class="centered">
+        <div class="emscripten" id="status">Downloading...</div>
+        <progress value="10" max="100" id="progress" hidden></progress>
+    </div>
+    <div id="container">
+        <canvas class="emscripten" id="canvas" oncontextmenu="event.preventDefault()" tabindex=-1></canvas>
+        <div class="emscripten fulscreen-container" id="fullscreen" style="display: none;">
+            <input type="button" value="Go Fullscreen" onclick="viewFullScreen()">
+        </div>
+    </div>
+    <script type='text/javascript'>
+        var canvasElement = document.getElementById('canvas');
+        var devicePixelRatio = window.devicePixelRatio || 1;
+
+        // Detect the visible width and height of the window
+        var canvasWidth = (screen.availWidth - (window.outerWidth - window.innerWidth)) * devicePixelRatio;
+        var canvasHeight = (screen.availHeight - (window.outerHeight - window.innerHeight)) * devicePixelRatio;
+
+        // Detect fullscreen change and resize canvas resolution accordingly
+        function fullscreenHandler(event) {
+            if (Module['JSCanvasSize']) {
+                if (isFullScreen()) {
+                    Module.JSCanvasSize(screen.width * devicePixelRatio, screen.height * devicePixelRatio, true, devicePixelRatio);
+                } else {
+                    if (document.referrer.includes('itch.io')) {
```

Comment:
This is oddly specific hack. Why is it needed?
It is worth documenting in comments, too.

Reply:
removed